demos/Secure-AI-inferencing-NGINX-IPU/demo/README.md

```diff
@@ -0,0 +1,214 @@
+forwards them to AI example pods running on the host.
+
+Run `./demo.sh predict`. This command calls the example [client script](./predict_images/run.py)
+which starts 20 parallel threads. Each thread
+will take a random picture of an animal and ask the NGINX loadbalancer to
+classify the picture. NGINX on the IPU will accept the HTTPS traffic and
+forward an unencrypted request to one of the demo pods on the host. Then the
+response is returned. Finally, the script prints the classification result
+of the images.
+
+See the NGINX configuration with the upstream pods via
+`./demo.sh oc_msh -n openshift-dpu-operator exec -ti pod/nginx -- cat /etc/nginx/conf.d/91-upstream.conf`.
+
+### Investigate the Demo Setup
+
+Now we can look at the system. Try some of the example commands.
+
+1. Look at the DPU operator's namespace on the OCP and microshift sides
+
+    - `./demo.sh oc_ocp -n openshift-dpu-operator get all`
+    - `./demo.sh oc_msh -n openshift-dpu-operator get all`
+
+1. Inspect the AI Inferencing Pods on the host
+
+    - `./demo.sh oc_ocp -n default get pod`
+    - `./demo.sh oc_ocp -n default get pod -o yaml`
+    - `./demo.sh oc_ocp -n default exec -ti pod/resnet50-model-server-1 -- bash`
+
+1. Inspect the NGINX loadbalancer Pod on the IPU
+
+    - `./demo.sh oc_msh -n openshift-dpu-operator get pod/nginx`
+    - `./demo.sh oc_msh -n openshift-dpu-operator exec -ti pod/nginx -- bash`
+
+1. Check IP configuration and traffic
+
+    Inside the pods, via `oc exec ...`, you can run `ip link`, `ip addr`,
+    `ping` and `tcpdump` to investigate what is happening.
+
+    Either start first an interactive shell via
+    `./demo.sh oc_msh -n openshift-dpu-operator exec -ti pod/nginx -- bash`,
+    or run as separate commands:
+
+    - `./demo.sh oc_msh -n openshift-dpu-operator exec -ti pod/nginx -- ps aux`
+    - `./demo.sh oc_msh -n openshift-dpu-operator exec -ti pod/nginx -- ip addr`
+    - `./demo.sh oc_msh -n openshift-dpu-operator exec -ti pod/nginx -- ping -c 3 10.56.217.12 # ping AI pod`
+    - `./demo.sh oc_msh -n openshift-dpu-operator exec -ti pod/nginx -- tcpdump -ni net2`
+
+### Run the Demo "inspect" Example
+
+Run `./demo.sh inspect`, which starts a tmx session with various panes. It will
+then execute a series of commands to step through the setup. This is an alternative
+to performing those steps yourself, as described [previously](#investigate-the-demo-setup).
+
+At the end of this "inspect" run, you end up with several tmux panes. On the right
+hand side you see tcpdump monitoring the traffic inside the AI pods and on the
+upper right pane you see the NGINX access log.
+
+On the upper left pane you can run `./demo.sh predict`. This triggers the client
+request that you can see in the NGINX access log and the network traffic forwarded
+to the pods.
+
+### Redeploy the NGINX Network Function and AI Pods
+
+The way to use Openshift is by creating custom resources. For example, to
```

Comment:
I would reword something like: "The way to use DPU Operator in Kubernetes is to create the CR (Custom Resource) following the fields and APIs from the CRD (Custom Resource Definition). The DPU Operator's manifests describe CRD and it's fields and APIs."